make check-static will fail on macOS  go 1.25

## Enhancement
error message



```cli

➜  tidb git:(master) make check-static
GOBIN=/Users/hyi/prs/tidb/tools/bin GO111MODULE=on go install github.com/golangci/golangci-lint/v2/cmd/golangci-lint@v2.4.0
GO111MODULE=on CGO_ENABLED=0 tools/bin/golangci-lint run -v $(go list ./... | sed 's|github.com/pingcap/tidb/||') --config .golangci.yml
INFO golangci-lint has version 2.4.0 built with go1.25.5 from (unknown, modified: ?, mod sum: "h1:qz6O6vr7kVzXJqyvHjHSz5fA3D+PM8v96QU5gxZCNWM=") on (unknown) 
INFO [config_reader] Used config file .golangci.yml 
INFO [config_reader] Module name "github.com/pingcap/tidb" 
INFO maxprocs: Leaving GOMAXPROCS=8: CPU quota undefined 
INFO [goenv] Read go env for 9.763833ms: map[string]string{"GOCACHE":"/Users/hyi/Library/Caches/go-build", "GOROOT":"/Users/hyi/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.25.5.darwin-arm64"} 
INFO [lintersdb] Active 17 linters: [bodyclose copyloopvar durationcheck errcheck gofmt goimports gosec ineffassign intrange lll makezero prealloc predeclared revive rowserrcheck staticcheck unused] 
INFO [loader] Go packages loading at mode 8767 (deps|imports|exports_file|files|name|types_sizes|compiled_files) took 37.373550958s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 309.884334ms 
INFO [linters_context/goanalysis] analyzers took 1m36.585701946s with top 10 stages: goimports: 25.35461279s, revive: 22.868076286s, buildir: 16.544017332s, gosec: 5.104987125s, gofmt: 3.804152288s, buildssa: 3.392083787s, unused: 1.349299255s, S1024: 1.111554085s, S1019: 938.242168ms, errcheck: 863.055951ms 
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package sigar: could not load export data: no export data for "github.com/cloudfoundry/gosigar" 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "third_party$" 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "builtin$" 
INFO [runner/exclusion_paths] Skipped 0 issues by pattern "examples$" 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "examples$", Linters: "gofmt, goimports"] 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "_test\\.go", Linters: "errcheck, gosec, makezero, rowserrcheck"] 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "third_party$", Linters: "gofmt, goimports"] 
INFO [runner/exclusion_rules] Skipped 0 issues by rules: [Path: "builtin$", Linters: "gofmt, goimports"] 
INFO [runner] processing took 25.043µs with stages: exclusion_paths: 14.041µs, exclusion_rules: 10.084µs, max_same_issues: 292ns, nolint_filter: 167ns, path_absoluter: 42ns, source_code: 42ns, max_from_linter: 42ns, path_prettifier: 42ns, diff: 42ns, fixer: 42ns, filename_unadjuster: 42ns, cgo: 42ns, invalid_issue: 41ns, severity-rules: 41ns, max_per_file_from_linter: 41ns, generated_file_filter: 0s, path_relativity: 0s, uniq_by_line: 0s, sort_results: 0s, path_shortener: 0s 
INFO [runner] linters took 9.316535042s with stages: goanalysis_metalinter: 9.316438084s 
ERRO Running error: can't run linter goanalysis_metalinter
buildir: failed to load package sigar: could not load export data: no export data for "github.com/cloudfoundry/gosigar" 
INFO Memory: 470 samples, avg is 348.2MB, max is 3044.9MB 
INFO Execution took 47.013435292s                 
make: *** [check-static] Error 3
```

--- a/Makefile
+++ b/Makefile
@@ -58,10 +58,17 @@ fmt: ## Format Go code using gofmt
 	@echo "gofmt (simplify)"
 	@gofmt -s -l -w -r 'interface{} -> any' $(FILES) 2>&1 | $(FAIL_ON_STDOUT)
 
+# On macOS, CGO is required for gosigar package to generate export data properly
+ifeq ($(GOOS),darwin)
+CGO_ENABLED_FOR_LINT := 1
+else
+CGO_ENABLED_FOR_LINT := 0
+endif
+
 .PHONY: check-static
 check-static: ## Run static code analysis checks
 check-static: tools/bin/golangci-lint
-	GO111MODULE=on CGO_ENABLED=0 tools/bin/golangci-lint run -v $$($(PACKAGE_DIRECTORIES)) --config .golangci.yml
+	GO111MODULE=on CGO_ENABLED=$(CGO_ENABLED_FOR_LINT) tools/bin/golangci-lint run -v $$($(PACKAGE_DIRECTORIES)) --config .golangci.yml
 
 .PHONY: check-file-perm
 check-file-perm:

